### ISSUE TITLE
DOWNLOAD_ONLY not parsed?

### ISSUE THREAD

**Issue by johelegp** [TheLartians/CPM.cmake#119]
I have the following setup, with _NEWLIB_ and its dependency _jegp_ both depending on _range-v3_.
Having translated directly from a sequence of `FetchContent`s and accompanying `FetchContent_MakeAvailable` to a sequence of `CPMFindPackage`, _jegp_ is first found and finds _range-v3_, but when _NEWLIB_ does it, `DOWNLOAD_ONLY` is `""`, as seen in the error below, for which I used [`cmake_print_variables`](https://cmake.org/cmake/help/latest/module/CMakePrintHelpers.html).

```
Running /usr/bin/cmake -GNinja <source-root>/newlib in <source-root>/newlib/build.
-- Found SFML 2.5.1 in /usr/lib64/cmake/SFML
-- CPM: adding package jegp_cmake_modules@0 (master)
-- CPM: adding package jegp@0 (master)
-- CMAKE_CURRENT_LIST_FILE="<source-root>/newlib/build/_deps/jegp-src/cmake/JEGPFindDependencies.cmake" ; CPM_ARGS_NAME="jegp_cmake_modules" ; DOWNLOAD_ONLY="NO"
-- CPM: jegp: adding package range-v3@0 (master)
-- CMAKE_CURRENT_LIST_FILE="<source-root>/newlib/tools/cmake/NEWLIBFindDependencies.cmake" ; CPM_ARGS_NAME="range-v3" ; DOWNLOAD_ONLY=""
CMake Error at tools/cmake/CPM.cmake:167 (cpm_fetch_package):
  cpm_fetch_package Function invoked with incorrect arguments for function
  named: cpm_fetch_package
Call Stack (most recent call first):
  tools/cmake/CPM.cmake:135 (CPMCheckIfPackageAlreadyAdded)
  tools/cmake/NEWLIBFindDependencies.cmake:21 (CPMFindPackage)
  CMakeLists.txt:11 (include)


-- CPM: using local package fmt@0
-- CPM: adding package mp-units@0 (master)
-- Configuring incomplete, errors occurred!
See also "<source-root>/newlib/build/CMakeFiles/CMakeOutput.log".
CMake process exited with exit code 1.
Elapsed time: 00:03.
```

I have circumvented this by first declaring my direct dependencies for which I know my other direct dependencies depend upon, which is necessary since the content is immediately made available, but it could become a nightmare in more complex scenarios.

**Comment by johelegp** [comment]
Here's JEGPFindDependencies.cmake: https://github.com/johelegp/jegp/blob/dd9e15ee4d5ec3e233d252fca091ee638777a8ce/cmake/JEGPFindDependencies.cmake.
And NEWLIBFindDependencies.cmake:
```CMake
if(NOT EXISTS ${CMAKE_CURRENT_LIST_DIR}/CPM.cmake)
    message(STATUS "Downloading CPM.cmake")
    file(DOWNLOAD
         https://github.com/TheLartians/CPM.cmake/raw/master/cmake/CPM.cmake
         ${CMAKE_CURRENT_LIST_DIR}/tools/CPM.cmake)
endif()

include(CPM)

find_package(SFML 2.5 REQUIRED COMPONENTS graphics)
CPMFindPackage(
    NAME jegp_cmake_modules
    GITHUB_REPOSITORY johelegp/jegp.cmake_modules
    GIT_TAG master
    GIT_SHALLOW True)
CPMFindPackage(
    NAME jegp
    GIT_REPOSITORY /mnt/sda5/documents/c++/repositories/jegp/jegp
    GIT_TAG master
    GIT_SHALLOW True)
CPMFindPackage(
    NAME range-v3
    GITHUB_REPOSITORY ericniebler/range-v3
    GIT_TAG master
    GIT_SHALLOW True
    DOWNLOAD_ONLY True)
if(range-v3_ADDED)
    add_library(range-v3::range-v3 INTERFACE IMPORTED GLOBAL)
    target_include_directories(range-v3::range-v3
                               INTERFACE ${range-v3_SOURCE_DIR}/include)
endif()
CPMFindPackage(
    NAME fmt
    GITHUB_REPOSITORY fmtlib/fmt
    GIT_TAG master
    GIT_SHALLOW True)
CPMFindPackage(
    NAME mp-units
    GITHUB_REPOSITORY mpusz/units
    GIT_TAG master
    GIT_SHALLOW True
    DOWNLOAD_ONLY True)

list(APPEND CMAKE_MODULE_PATH ${jegp_cmake_modules_SOURCE_DIR})

if(mp-units_ADDED)
    add_subdirectory(${mp-units_SOURCE_DIR}/src ${mp-units_BINARY_DIR})
endif()
```
The dependency story of _jegp_cmake_modules_ is the same as that of _range-v3_, but the former works. The only difference is that _range-v3_ is declared as `DOWNLOAD_ONLY True`.

**Comment by TheLartians** [comment]
Hey thanks for the report! I think the problem was that we were incorrectly trying to download the package when we should have only been checking its availability. I've corrected that in #120. Could you test if [the patched version](https://raw.githubusercontent.com/TheLartians/CPM.cmake/4c459aff6fb3996c2379e048585be67623f98b05/cmake/CPM.cmake) solves the issue for you?

**Comment by johelegp** [comment]
It does solve it.

**Comment by TheLartians** [comment]
Great, thanks for verifying it! The fix is now published in version 0.24.
